### ISSUE TITLE
Tftp server resource shouldn't server until CheckApply runs successfully

### ISSUE THREAD

**Issue by purpleidea** [purpleidea/mgmt#577]
Basically we're in the DAG, so make sure we've let all the other resources (like files) get built (that might be used for the file server) before the server sends anything out. We could either block or just not return anything yet.

As an aside, does the rf.ReadFrom bit not get cancelled properly on close??

**Comment by hariTiwari442** [comment]
Hey!
Please assign this to me!
Thanks.

**Comment by purpleidea** [comment]
No need to assign it, just send a patch, nobody is working on this at the moment! Cheers!

**Comment by hariTiwari442** [comment]
Hi I read introductory Blog related to the mgmt and your TODO list, Although I am understanding the whole project but to understand whole code and flow and logic, Can you please create some low difficulty level and easy fix issue so that I can learn the code base while solving the easy issue.

**Comment by purpleidea** [comment]
Yes I can-- can you please give me a rough idea of your golang experience level, and how many hours you'd like to initially devote to the project? This was I can help with something more appropriate for you. Cheers

**Comment by hariTiwari442** [comment]
Hi currently i am working in golang from last 4 months but previously i working on different web technologies like Node and others, i am also done competitive programming with with c++ and when i saw your DAG explanation i really got excited to know this project more

**Comment by purpleidea** [comment]
@hariTiwari442 Perhaps you could try implementing a simple mgmt function, like `is_virtual` for example?

It's challenging to find beginner golang projects, but I think this might be a good bet. Alternatively implement a simple resource like sysctl or something else you have in mind.

**Comment by hariTiwari442** [comment]
Hi, Thanks for your guide,
Sure once I complete the code base, I will start with simple functions and resources thanks.

**Comment by purpleidea** [comment]
Fix merged! \o/
